Use stack for dependency management

gfontenot opened the pull request.

gfontenot committed:
Use stack for dependency management

gabebw commented on README.md:
```diff
@@ -15,13 +15,12 @@ where the static site is built, and then the whole thing is deployed to
 % brew update
-% brew install ghc cabal-install
-% cabal install cabal-install
+% brew install ghc stack
```
- The package is named `haskell-stack`, not `stack`
- `haskell-stack` depends on `ghc`, so you probably don't need to explicitly mention `ghc`

This line could thus be: `% brew install haskell-stack`

gfontenot committed:
Install stack on Circle

gfontenot committed:
stack is under the package name haskell-stack

gfontenot committed:
Circle uses linux you idiot

gfontenot committed:
Manually install stack

gfontenot committed:
Update ghc version

gfontenot committed:
use stack setup on CI

gfontenot committed:
Fix circle

no idea but pat is doing it so ¯\_(ツ)_/¯

gfontenot committed:
Try to fix build

gfontenot committed:
Remove explicit ghc version from circle

This is controlled by Stack now

gfontenot committed:
Run stack build as part of the test process

Not sure why this wasn't already happening

pbrisbin commented on README.md:
```diff
@@ -15,13 +15,12 @@ where the static site is built, and then the whole thing is deployed to
 ### Dependencies ###
 
-You'll need to have `cabal` and `ghc` installed. If you're on OS X, you can do
+You'll need to have `stack` and `ghc` installed. If you're on OS X, you can do
```
Are you not using a stack-installed GHC?

gfontenot replied:
Should I just _always_ be using stack to install ghc?

pbrisbin replied:
[I do](https://github.com/pbrisbin/dotfiles/blob/master/tag-haskell/stack/config.yaml#L4).

pbrisbin commented on circle.yml:
```diff
@@ -1,16 +1,28 @@
-    - cabal run --  build
-    - cabal run --  check --internal-links
+    - stack build
+    - stack exec site -- clean
```
Probably don't need this. Do you expect `_site` to have content in a fresh checkout?

gfontenot replied:
Nope! Good point.

pbrisbin commented:
LGTM

gfontenot committed:
Don't clean before building

This is a fresh checkout, so `_site_` shouldn't actually have any
content yet. That makes this line irrelevant.

gfontenot committed:
Don't call out the need for ghc

Stack handles everything! It also won't take a while anymore!

gfontenot committed:
Move stack setup to bin/setup

This seems reasonable to do on setup in general, not just for CI.
